dspace-api/src/test/java/org/dspace/content/dao/RelationshipDAOImplTest.java

```diff
@@ -0,0 +1,186 @@
+
+    private EntityType entityTypeOne;
+
+    private EntityType entityTypeTwo;
+
+    protected CommunityService communityService = ContentServiceFactory.getInstance().getCommunityService();
+    protected CollectionService collectionService = ContentServiceFactory.getInstance().getCollectionService();
+    protected ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+    protected InstallItemService installItemService = ContentServiceFactory.getInstance().getInstallItemService();
+    protected WorkspaceItemService workspaceItemService = ContentServiceFactory.getInstance().getWorkspaceItemService();
+    protected RelationshipTypeService relationshipTypeService =
+            ContentServiceFactory.getInstance().getRelationshipTypeService();
+    protected RelationshipService relationshipService = ContentServiceFactory.getInstance().getRelationshipService();
+    protected EntityTypeService entityTypeService = ContentServiceFactory.getInstance().getEntityTypeService();
+
+    /**
+     * Initalize DSpace objects used for testing for each test
+     */
+    @Before
+    @Override
+    public void init() {
+        super.init();
+        try {
+            // Create objects for testing
+            context.turnOffAuthorisationSystem();
+            owningCommunity = communityService.create(null, context);
+            collection = collectionService.create(context, owningCommunity);
+            WorkspaceItem workspaceItem = workspaceItemService.create(context, collection, false);
+            WorkspaceItem workspaceItemTwo = workspaceItemService.create(context, collection, false);
+            itemOne = installItemService.installItem(context, workspaceItem);
+            itemTwo = installItemService.installItem(context, workspaceItemTwo);
+            itemService.addMetadata(context, itemOne, "relationship", "type", null, Item.ANY, "Publication");
+            itemService.addMetadata(context, itemTwo, "relationship", "type", null, Item.ANY, "Person");
+            itemService.update(context, itemOne);
+            itemService.update(context, itemTwo);
+            entityTypeOne = entityTypeService.create(context, "Person");
+            entityTypeTwo = entityTypeService.create(context, "Publication");
+            relationshipType = relationshipTypeService.create(context, entityTypeTwo, entityTypeOne,
+                    "isAuthorOfPublication", "isPublicationOfAuthor",0,10,0,10);
+            relationship = relationshipService.create(context, itemOne, itemTwo, relationshipType, 0, 0);
+            relationshipService.update(context, relationship);
+            relationshipsList.add(relationship);
+            context.restoreAuthSystemState();
+        } catch (Exception e) {
+            log.error(e);
+            fail(e.getMessage());
+        }
+    }
+
+    /**
+     * Delete all initalized DSpace objects after each test
+     */
+    @After
+    @Override
+    public void destroy() {
```

Comment:
If you use Builder classes, you can remove this entire `destroy()` method, as the Builders automatically track all content created and delete the content after the test runs.